### ISSUE TITLE
Type checking bugs in `inf`

### ISSUE THREAD

**Issue by alexfeld** [marionmari/pyGPs#27]
There are 5 type checking bugs that are causing optimizer failures in `inf.py` At lines 98, 100, 102, 360, and 363. The bugs are all the same type. The current lines are

```python
if <variable>:
    <do action>
```

These should read

```python
if <variable> is not None:
    <do action>
```

The first method fails if the variable is a `numpy.ndarray`.

**Comment by alexfeld** [comment]
I've added a fix for this in #29

**Comment by marionmari** [comment]
Great! Thanks!
